rand_core/src/lib.rs

```diff
@@ -95,12 +95,27 @@ pub trait Rng {
     /// Fill `dest` entirely with random data.
     ///
     /// This method does *not* have any requirement on how much of the
+    /// generated random number stream is consumed; e.g. `fill_bytes_via_u64`
+    /// implementation uses `next_u64` thus consuming 8 bytes even when only
+    /// 1 is required. A different implementation might use `next_u32` and
+    /// only consume 4 bytes; *however* any change affecting *reproducibility*
+    /// of output must be considered a breaking change.
+    fn fill_bytes(&mut self, dest: &mut [u8]);
+
+    /// Fill `dest` entirely with random data.
+    ///
+    /// If a RNG can encounter an error, this is the only method that reports
+    /// it. The other methods either handle the error, or panic.
+    ///
+    /// This method does *not* have any requirement on how much of the
     /// generated random number stream is consumed; e.g. `try_fill_via_u64`
     /// implementation uses `next_u64` thus consuming 8 bytes even when only
     /// 1 is required. A different implementation might use `next_u32` and
     /// only consume 4 bytes; *however* any change affecting *reproducibility*
     /// of output must be considered a breaking change.
-    fn try_fill(&mut self, dest: &mut [u8]) -> Result<(), Error>;
+    fn try_fill(&mut self, dest: &mut [u8]) -> Result<(), Error> {
+        Ok(self.fill_bytes(dest))
+    }
```

Comment:
Default impl... I'd like not to have this

Reply:
Maybe we can discuss it in https://github.com/dhardy/rand/issues/12? Personally I don't see a problem, but I may be missing something.

Reply:
There's an [argument in the RFC about this](https://github.com/dhardy/rfcs/blob/rand-core/text/0000-rand-core.md#default-implementations-of-functions)

Reply:
I will remove it